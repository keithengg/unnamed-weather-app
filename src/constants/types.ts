export type Weather = {
    temp: number;
    temp_max: number;
    temp_min: number;
    humidity: number;
    city: string;
    time: string;
    cloud_status?: string;
}