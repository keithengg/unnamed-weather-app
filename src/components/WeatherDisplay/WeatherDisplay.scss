.weather-display {
    &__container {
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-template-rows: 1fr;
    }

    &__left-container {
        display: flex;
        flex-direction: column;
        gap: 4px
    }

    &__right-container {
        display: flex;
        flex-direction: column;
        justify-content: flex-end;
        align-items: flex-end;
        gap: 4px;
    }

    &__temp {
        color: #6940b9;
        font-weight: bold;
        font-size: 64px;
        line-height: 100%;
    }

    &__city {
        font-weight: bold;
    }

    &__image {
        position: absolute;
        top: -120px;
        right: -20px;
        height: 220px;

        @media only screen and (min-width: 767px) {
            top: -160px;
            right: 0;
            height: 320px;
        }
    }
}