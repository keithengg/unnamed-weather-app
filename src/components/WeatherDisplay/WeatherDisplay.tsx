import React from "react";
import { Weather } from '../../constants/types'
import './WeatherDisplay.scss';

const NightImage = require('../../assets/cloud.png');
const SunImage = require('../../assets/sun.png');

interface Props {
    weatherData: Weather
};

const WeatherDisplay: React.FC<Props> = ({ weatherData }) => {
    const { temp, temp_max, temp_min, city, humidity, cloud_status, time } = weatherData;

    const isDay = new Date().getHours() <= 17; // Check if it's before 5:00 PM

    return (
        <div className="relative">
            <h1>Today's Weather</h1>
            <div className="weather-display__container">
                <div className="weather-display__left-container">
                    <h2 className="weather-display__temp">{temp}&deg;</h2>
                    <div className="flex gap-x-2">
                        {temp_max && <p className="weather-display__temp-max">H: {temp_max}&deg;</p>}
                        {temp_min && <p className="weather-display__temp-min">L: {temp_min}&deg;</p>}
                    </div>
                    <h3 className="weather-display__city">{city}</h3>
                </div>
                <div className="weather-display__right-container">
                    <p className="weather-display__cloud">{cloud_status}</p>
                    <p className="weather-display__humidity">Humidity: {humidity}%</p>
                    <p className="weather-display__time">{time}</p>
                </div>
            </div>
            <img
                src={isDay ? SunImage : NightImage}
                className="weather-display__image"
                alt={isDay ? 'Graphical Image of a sun behind a cloud' : 'Graphical Image of a cloud'}
            />
        </div>
    )
}

export default WeatherDisplay;