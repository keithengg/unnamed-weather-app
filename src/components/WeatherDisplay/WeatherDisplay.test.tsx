import React from "react";
import { render } from "@testing-library/react";
import { Weather } from '../../constants/types';
import WeatherDisplay from "./WeatherDisplay";

describe("<WeatherDisplay />", () => {
    const mockWeatherData: Weather = {
        temp: 100,
        temp_max: 100,
        temp_min: 100,
        humidity: 100,
        city: "name, country",
        cloud_status: "cloudy",
        time: ""
    }

    it('should render correctly and match snapshot', () => {
        const { container } = render(<WeatherDisplay weatherData={mockWeatherData} />)
        expect(container).toMatchSnapshot()
    });
});