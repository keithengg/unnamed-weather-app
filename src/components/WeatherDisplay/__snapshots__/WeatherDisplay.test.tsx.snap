// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherDisplay /> should render correctly and match snapshot 1`] = `
<div>
  <div
    class="relative"
  >
    <h1>
      Today's Weather
    </h1>
    <div
      class="weather-display__container"
    >
      <div
        class="weather-display__left-container"
      >
        <h2
          class="weather-display__temp"
        >
          100
          °
        </h2>
        <div
          class="flex gap-x-2"
        >
          <p>
            H: 
            100
            °
          </p>
          <p>
            L: 
            100
            °
          </p>
        </div>
        <h3
          class="weather-display__city"
        >
          name, country
        </h3>
      </div>
      <div
        class="weather-display__right-container"
      >
        <p>
          cloudy
        </p>
        <p>
          Humidity: 
          100
          %
        </p>
        <p />
      </div>
    </div>
    <img
      alt="Graphical Image of a sun behind a cloud"
      class="weather-display__image"
      src="sun.png"
    />
  </div>
</div>
`;
